### ISSUE TITLE
Docker save errors when using sha256 imageId

### ISSUE THREAD

**Issue by marcosnils** [docker/docker#21375]
**Output of `docker version`:**

```
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 21:49:11 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 21:49:11 2016
 OS/Arch:      linux/amd64

```

**Output of `docker info`:**

```
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 38
Server Version: 1.10.3
Storage Driver: aufs
 Root Dir: /home/marcos/docker/aufs
 Backing Filesystem: extfs
 Dirs: 298
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.19.0-51-generic
Operating System: Ubuntu 15.04
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.704 GiB
Name: XPS
ID: LHF6:NIMP:TNYU:QXAA:KXZP:RRV5:MJUB:GIZG:GZGQ:MFE3:H3WN:WPN7
Debug mode (server): true
 File Descriptors: 19
 Goroutines: 36
 System Time: 2016-03-21T18:54:46.651897632-03:00
 EventsListeners: 0
 Init SHA1: 
 Init Path: /usr/bin/docker
 Docker Root Dir: /home/marcos/docker
Username: marcosnils
Registry: https://index.docker.io/v1/

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1.docker images --no-trunc
```
REPOSITORY                             TAG                 IMAGE ID                                                                  CREATED             SIZE
mantika/moderations-landing            master              sha256:7ea5c2193566e60519813e317184c2dd2dac367ba13e22707e3fe7ba8dd716ba   About an hour ago   256 MB
```
2.
docker save 7ea5c2193566 >> Works 
docker save 7ea5c2193566e60519813e317184c2dd2dac367ba13e22707e3fe7ba8dd716ba >> Errors

**Describe the results you received:**
Error response from daemon: Invalid repository name (7ea5c2193566e60519813e317184c2dd2dac367ba13e22707e3fe7ba8dd716ba), cannot specify 64-byte hexadecimal strings

**Describe the results you expected:**
As it works with the short imageId i'd expect to work with the long format.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by thaJeztah** [comment]
Thanks for reporting; I think this is similar to https://github.com/docker/docker/issues/20972, and *may* be resolved by https://github.com/docker/docker/pull/21002, but I'm not sure if that also fixes this issue for `docker save`, so

/cc @tonistiigi

**Comment by marcosnils** [comment]
@thaJeztah seems like it will be resolved by #21002 per https://github.com/docker/docker/pull/21002/files#diff-0e1bc2b0323f2b28cefb8eac65ca152c

Thx!.

**Comment by thaJeztah** [comment]
Oh! You're right, thanks!
